pkg/cmd/cli/restore/describe.go

```diff
@@ -17,18 +17,181 @@ limitations under the License.
+				}
+			} else {
+				restores, err = arkClient.ArkV1().Restores(api.DefaultNamespace).List(listOptions)
+				cmd.CheckError(err)
+			}
+
+			first := true
+			for _, restore := range restores.Items {
+				s := output.Describe(func(out io.Writer) {
+					describeRestore(out, &restore, arkClient)
+				})
+				if first {
+					first = false
+					fmt.Print(s)
+				} else {
+					fmt.Printf("\n\n%s", s)
+				}
+			}
+			cmd.CheckError(err)
 		},
 	}
 
+	c.Flags().StringVarP(&listOptions.LabelSelector, "selector", "l", listOptions.LabelSelector, "only show items matching this label selector")
+
+	output.BindFlags(c.Flags())
+
 	return c
 }
+
+func describeRestore(out io.Writer, restore *api.Restore, arkClient clientset.Interface) {
+	output.DescribeMetadata(out, restore.ObjectMeta)
+
+	fmt.Fprintln(out)
+	fmt.Fprintf(out, "Backup:\t%s\n", restore.Spec.BackupName)
+
+	fmt.Fprintln(out)
+	fmt.Fprintf(out, "Namespaces:\n")
+	var s string
+	if len(restore.Spec.IncludedNamespaces) == 0 {
+		s = "*"
+	} else {
+		s = strings.Join(restore.Spec.IncludedNamespaces, ", ")
+	}
+	fmt.Fprintf(out, "\tIncluded:\t%s\n", s)
+	if len(restore.Spec.ExcludedNamespaces) == 0 {
+		s = "<none>"
+	} else {
+		s = strings.Join(restore.Spec.ExcludedNamespaces, ", ")
+	}
+	fmt.Fprintf(out, "\tExcluded:\t%s\n", s)
+
+	fmt.Fprintln(out)
+	fmt.Fprintf(out, "Resources:\n")
+	if len(restore.Spec.IncludedResources) == 0 {
+		s = "*"
+	} else {
+		s = strings.Join(restore.Spec.IncludedResources, ", ")
+	}
+	fmt.Fprintf(out, "\tIncluded:\t%s\n", s)
+	if len(restore.Spec.ExcludedResources) == 0 {
+		s = "<none>"
+	} else {
+		s = strings.Join(restore.Spec.ExcludedResources, ", ")
+	}
+	fmt.Fprintf(out, "\tExcluded:\t%s\n", s)
+	var clusterResources string
+	switch {
+	case restore.Spec.IncludeClusterResources == nil:
+		clusterResources = "auto"
+	case *restore.Spec.IncludeClusterResources == true:
+		clusterResources = "included"
+	case *restore.Spec.IncludeClusterResources == false:
+		clusterResources = "excluded"
+	}
+	fmt.Fprintf(out, "\tCluster-scoped:\t%s\n", clusterResources)
+
+	fmt.Fprintln(out)
+	output.DescribeMap(out, "Namespace mappings", restore.Spec.NamespaceMapping)
+
+	fmt.Fprintln(out)
+	s = "<none>"
+	if restore.Spec.LabelSelector != nil {
+		s = metav1.FormatLabelSelector(restore.Spec.LabelSelector)
+	}
+	fmt.Fprintf(out, "Label selector:\t%s\n", s)
+
+	fmt.Fprintln(out)
+	b := true
+	if restore.Spec.RestorePVs != nil {
+		b = *restore.Spec.RestorePVs
+	}
+	s = fmt.Sprintf("%t", b)
+	fmt.Fprintf(out, "Restore PVs:\t%s\n", s)
+
+	fmt.Fprintln(out)
+	fmt.Fprintf(out, "Phase:\t%s\n", restore.Status.Phase)
+
+	fmt.Fprintln(out)
+	fmt.Fprint(out, "Validation errors:")
+	if len(restore.Status.ValidationErrors) == 0 {
+		fmt.Fprintf(out, "\t<none>\n")
+	} else {
+		for _, ve := range restore.Status.ValidationErrors {
+			fmt.Fprintf(out, "\t%s\n", ve)
+		}
+	}
+
+	fmt.Fprintln(out)
+	describeRestoreResults(out, arkClient, restore.Name)
```

Comment:
maybe optimize the case where there are 0 errors/warnings?

Reply:
Good idea, I'll do that tomorrow.